File: src/main/java/com/practica/crean/arquitecture/shared/excepciones/ResourceNotFoundException.java
package com.practica.crean.arquitecture.shared.excepciones;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseBody
@ResponseStatus(HttpStatus.NOT_FOUND)
public class ResourceNotFoundException extends RuntimeException {

    public ResourceNotFoundException(String mensaje){
        super(mensaje);
    }

}

File: src/main/kotlin/com/practica/crean/arquitecture/shared/excepciones/ResourceNotFoundException.kt
package com.example.demo.shared.excepciones

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseBody
@ResponseStatus(HttpStatus.NOT_FOUND)
class ResourceNotFoundException(mensaje : String) : RuntimeException(mensaje)